src/validation.cpp

```diff
@@ -1467,11 +1471,16 @@ MempoolAcceptResult MemPoolAccept::AcceptSingleTransaction(const CTransactionRef
                                             ws.m_base_fees, effective_feerate, single_wtxid);
     }
 
-    if (!FinalizeSubpackage(args, workspaces)) {
-        // The only possible failure reason is fee-related (mempool full).
-        // Failed for fee reasons. Provide the effective feerate and which txns were included.
-        Assume(ws.m_state.GetResult() == TxValidationResult::TX_RECONSIDERABLE);
-        return MempoolAcceptResult::FeeFailure(ws.m_state, CFeeRate(ws.m_modified_fees, ws.m_vsize), {ws.m_ptx->GetWitnessHash()});
+    FinalizeSubpackage(args);
+
+    // Limit the mempool, if appropriate.
+    if (!args.m_bypass_limits) {
```

Comment:
Agreed, I have restored this argument where `LimitMempoolSize()` is invoked in `AcceptSingleTransaction()`, and I pulled this change out into its own commit so that it's easier to review and less tangled with the other changes.

More thoughts on this [here](https://github.com/bitcoin/bitcoin/pull/28251#issuecomment-2433501732).  It would be nice to have a test that packages which involve a parent transaction that would be just at the bottom of the mempool without their children are able to make it in with a child, if the resulting package would score higher in the mempool than other transactions.

Reply:
I'm working on a test case for master for that case

Reply:
I think this is all good now (your test in #31152 passes for me locally), so marking this as resolved.

Reply:
nice catch